### ISSUE TITLE
Insight in the regularization to improve the continuity and consistency of predicted SDF values.

### ISSUE THREAD

**Issue by EugeneLiu01** [nexuslrf/SPIDR#10]
Hi! Thanks for your brilliant work!
I am wondering the insight in `the regularization to improve the continuity and consistency of predicted SDF values`([code](https://github.com/nexuslrf/SPIDR/blob/0dfe219dec65e60f4b0799075e496b2a9db8de14/models/base_rendering_model.py#L850) here i guess), which correspoding the Eq. 11

**Comment by nexuslrf** [comment]
Hi, thanks for your interest!
A figure from my old slides may help you better understand the design of this regularization.

![image](https://github.com/nexuslrf/SPIDR/assets/34814176/565eb294-bc27-4a47-8a99-0a8a7a9d49bc)

the change in the predicted SDF  $\Delta d$ is constrained in such a right triangle relationship.

**Comment by EugeneLiu01** [comment]
Thank you for your sharing!
It looks like that this regularization might has a same effect as eikonal loss, which constrain the gradient modulus of the SDF is one $(\| \nabla SDF \| - 1)^2 = 0$, while eikonal loss is a stronger one?

**Comment by nexuslrf** [comment]
Yeah, your understanding is correct. This loss term can be treated as a loosened approximation of eikonal loss. We use this to replace the eikonal term to avoid second-order optimization wrt SDF gradient.

**Comment by EugeneLiu01** [comment]
I am wonder whether this term also benifit to the smoothness of sdf, because the discretized neural representation based on point clouds is unstructured, and it is extremely likely that the sdf will not be smooth.

**Comment by nexuslrf** [comment]
From my observation, it can improve the smoothness (also shown in Fig. 3 of the paper), but it cannot guarantee perfect smoothness due to the nature of unstructured points. Anyway, this work is just an earlier attempt. I feel an eikonal loss with high-order numerical gradients (e.g., Neuralangelo) could be more beneficial. 
I guess some similar thoughts could be applied to the SDF reconstruction with gaussian spatts. XD).

**Comment by EugeneLiu01** [comment]
Constructing neural surface from point indeed is a hard problem. Thanks for your great work taking a step forward to this problem!
